app/controllers/course_notes_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class CourseNotesController < ApplicationController
+  before_action :set_course_note, only: [:find_course_note, :update, :destroy]
+
+  def show
+    course_notes = CourseNote.where(courses_id: params[:id])
+    render json: { courseNotes: course_notes }
+  end
+
+  def find_course_note
+    render json: { courseNote: @course_note }
+  end
+
+  def update
+    if @course_note.update_note(course_note_params)
+      render json: { success: true }
+    else
+      render json: { error: 'Failed to update course note' }, status: :unprocessable_entity
+    end
+  end
+
+  def create
+    note_details = CourseNote.new.create_new_note(course_note_params)
```

Comment:
This is not the idiomatic Rails way to create a new record. If you do `CourseNote.create(attrributes)` instead, it will return a CourseNote note object which you can check for validation errors (by calling `.errors` on it). Or you can do `CourseNote.create!` and it will raise those errors directly so that you can rescue them and return an appropriately detailed error response.